opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -255,11 +259,78 @@ public Boolean checkIfExists(Table table, String baseEntityId, SQLiteDatabase sq
         return false;
     }
 
+    private List<String> getFormSubmissionIdsFromJsonArray(JSONArray array) {
+        if (array != null && array.length() != 0) {
+            List<String> stringList = new ArrayList<>();
+            for (int i = 0; i < array.length(); i++) {
+                JSONObject jsonObject = array.optJSONObject(i);
+                if (jsonObject != null) {
+                    String formSubmissionId = jsonObject.optString(event_column.formSubmissionId.name());
+                    if (StringUtils.isNotBlank(formSubmissionId)) {
+                        stringList.add(formSubmissionId);
+                    }
+                }
+            }
+            return stringList;
+        }
+        return new ArrayList<>();
+    }
+
+    protected void populateFormSubmissionIds(@NonNull List<String> formSubmissionIdsList,
+                                             Set<String> formSubmissionIds) {
+        populateFormSubmissionIds(formSubmissionIdsList, formSubmissionIds, null);
+    }
+
+    protected void populateFormSubmissionIds(@NonNull List<String> formSubmissionIdsList,
+                                             Set<String> formSubmissionIds, @Nullable Integer pageSize) {
```

Comment:
not sure can we remove parameter `@Nullable Integer pageSize` and use `FORM_SUBMISSION_IDS_PAGE_SIZE`